odsApi_versioned_docs/version-7.3/technical-articles/authorizing-requests-using-custom-database-views.md

```diff
@@ -0,0 +1,179 @@
+authorization strategies that limit an API client’s access to the following
+groups of Students:
+
+* Students enrolled in CTE courses
+* Students in the Gifted and Talented program
+* Students in preschool or early childhood education
+
+The view-based support is not limited to Students however and can use any Ed-Fi
+or extension entity as the basis for authorization. As additional contrived
+examples, one could implement authorization strategies as follows:
+
+* AssessmentWithReadingInTheIdentifier - Limit access to Assessment-related data
+  (including Assessment metadata and/or Student Assessment results) for items
+  where the associated Assessment has an AssessmentIdentifier containing the
+  text “Reading”.
+* TransportationTypeDescriptorWithABus - Limit access to Student Transporation
+  resource items with a TransportationTypeDescriptor code value that contains
+  the word “Bus”.
+
+The approach provides tremendous flexibility but requires that implementers
+adhere to strict conventions when creating the authorization strategy names and
+the corresponding database views. The process and required conventions are
+described below.
+
+## Step 1: Identify the Authorization Strategy
+
+A custom view-based authorization strategy starts with the selection of a basis
+entity. This is the entity whose primary key values must be present on the root
+entity of the resources being authorized. In the existing relationship-based
+authorization strategies, the basis entities are limited to types of people
+(Student/Staff/Parent/Contact) and/or education organizations. With custom
+view-based authorization strategies, any entity in the model (including
+extensions) can be used.
+
+For the sake of an example, let’s build an authorization strategy based on
+Assessment that can be used to authorize access to dependent resources,
+filtering items to those referencing Assessments that contain an academic
+subject of “Mathematics” (see database diagram below for reference).
+
+## Step 2: Create the ODS Database View
+
+### Identify the Basis Entity/Table
+
+The database view built for the authorization strategy must contain the basis
+table’s primary key values because these will be used by the authorization
+strategy implementation to join to the subject tables being authorized.
+
+### Plan the Query
+
+In the case of Assessment table, the primary key consists of the
+AssessmentIdentifier and Namespace columns. While these two columns determine
+the query results that must be produced by the custom authorization view
+definition, they don’t necessarily need to come directly from the basis table.
+
+In this example, we are actually interested in filtering Assessments based on
+the presence of the Academic Subject of “Mathematics”. This value is stored in
+the AcademicSubjectDescriptorId column of the AssessmentAcademicSubject child
+table:
+
+![Database Diagram](/img/reference/ods-api/Assessment-AssessmentAcademicSubject-Diagram.webp)
```

Comment:
Most of the database diagrams are converted to mermaid diagrams. e.g. https://github.com/Ed-Fi-Alliance-OSS/ed-fi-alliance-oss.github.io/blob/main/odsApi_versioned_docs/version-7.3/platform-dev-guide/configuration/context-based-routing-for-year-specific-ods.md?plain=1#L53. Can you convert this one?

Reply:
This is now converted to mermaid.